lib/model/typing_status.dart

```diff
@@ -0,0 +1,78 @@
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+
+import '../api/model/events.dart';
+import 'narrow.dart';
+
+/// The model for tracking the typing status organized by narrows.
+class TypingStatus extends ChangeNotifier {
```

Comment:
Consumers of the typing
    status will only be notified if a typist has been added or removed from
    any of the narrows.

Similarly to my comment just above (or a bit below if reading this in the Files tab) about:
https://github.com/zulip/zulip-mobile/blob/main/docs/style.md#commit-messages-vs-comments

In this case the information is something one should know when consuming this class, so it's good to put in its doc:

```suggestion
/// The model for tracking the typing status organized by narrows.
///
/// Listeners are notified when a typist is added or removed from any narrow.
class TypingStatus extends ChangeNotifier {
```